docs/deployment.md

```diff
@@ -1,66 +1,182 @@
 # Deployment
 
-One of the key features of BigFlow CLI is the full automation of the build and deployment process.
-BigFlow can [build](project_setup_and_build.md) your workflows
-to dockerized Airflow DAGs and deploy them to Google Cloud Composer.
+One of the key features of BigFlow is the full automation of the build and deployment process.
+BigFlow can dockerize your workflows and deploy them to Google Cloud Composer.
 
-## Deployment artifacts
-There are two deployment artifacts:
+## GCP runtime environment
+
+BigFlow GCP runtime environment consists of two services:
+
+1. Google [Cloud Composer](#cloud-composer),
+2. [Docker Registry](#docker-registry).
+
+Typically, for one software project, teams use one or more
+GCP projects (for dev, test, and prod environments)
+and one long-running Composer instance per each GCP project.
+
+Docker images are heavy files, so pushing them only once to GCP greatly
+reduces subsequent deployment time (it's safe because images are immutable).
+That's why we recommend using a single Docker Registry instance of,
+shared by all environments. 
+  
+There are two [deployment artifacts](project_setup_and_build.md#deployment-artifacts):
 
 1. Airflow DAG files with workflows definitions,
-1. Docker image with workflows computation code.
+1. a Docker image with workflows computation code.
 
-## GCP runtime environment
+During deployment, BigFlow uploads your DAG files to Composer's [DAGs folder](#composers-dags-folder)
+and pushes your Docker image to Docker Registry.
+
+Read more about deployment artifacts in [Project setup and build](project_setup_and_build.md).
+
+## Cloud Composer
+
+Shortly speaking, a Cloud Composer is Airflow-as-a-Service.
+
+Unfortunately for Python users, Composer's architecture is flawed
+because Python libraries required by DAGs have to be
+[installed manually](https://cloud.google.com/composer/docs/how-to/using/installing-python-dependencies) on Composer.
+To make it worse, installing dependencies forces a Composer instance to restart.
+It not only takes time but often fails. In the worst scenario, you need to spawn a new Composer instance.
```

Comment:
often fails because of dependencies clash - coś takiego, trzeba podać jakiś powód dlaczego failuje